src/net/atlas/db.rs

```diff
@@ -0,0 +1,379 @@
+use rusqlite::types::ToSql;
+use rusqlite::Row;
+use rusqlite::Transaction;
+use rusqlite::{Connection, OpenFlags, NO_PARAMS};
+
+use std::collections::HashSet;
+use std::convert::From;
+use std::convert::TryFrom;
+use std::fs;
+
+use util::db::tx_begin_immediate;
+use util::db::DBConn;
+use util::db::Error as db_error;
+use util::db::{query_count, query_int, query_row, query_rows, u64_to_sql, FromColumn, FromRow};
+
+use util;
+use util::hash::{bin_bytes, hex_bytes, to_bin, to_hex, Hash160};
+use util::log;
+use util::macros::is_big_endian;
+use util::secp256k1::Secp256k1PrivateKey;
+use util::secp256k1::Secp256k1PublicKey;
+
+use vm::types::QualifiedContractIdentifier;
+
+use chainstate::burn::{BlockHeaderHash, ConsensusHash};
+use chainstate::stacks::StacksBlockId;
+use net::StacksMessageCodec;
+
+use super::{Attachment, AttachmentInstance};
+
+pub const ATLASDB_VERSION: &'static str = "23.0.0.0";
+
+const ATLASDB_SETUP: &'static [&'static str] = &[
+    r#"
+    CREATE TABLE attachments(
+        hash TEXT UNIQUE PRIMARY KEY,
+        content BLOB NOT NULL
+    );"#,
+    r#"
+    CREATE TABLE attachment_instances(
+        id INTEGER PRIMARY KEY AUTOINCREMENT,
+        content_hash TEXT,
+        created_at INTEGER NOT NULL,
+        consensus_hash STRING NOT NULL,
+        block_header_hash STRING NOT NULL,
+        block_id STRING NOT NULL,
+        position_in_page INTEGER NOT NULL,
+        page_index INTEGER NOT NULL,
+        block_height INTEGER NOT NULL,
+        is_available INTEGER NOT NULL,
+        metadata STRING NOT NULL,
+        contract_id STRING NOT NULL
+    );"#,
+    r#"
+    CREATE TABLE inboxed_attachments(
```

Comment:
Can you add some docstrings somewhere to explain what the difference is between `inboxed_attachments` and `attachments`, and why we need two tables for them, as well as why we need code to move `content` from one table to another?  I'm not following the logic or reasoning behind this.

In general, I'd try to avoid moving `content` between tables -- the content blob can be somewhat large.  Why not just store it to one table, and reference it from others via a `JOIN`?

Reply:
Yeah I agree and had the same concern. Will revisit the design.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/2013/commits/00769f8f9a3c5279ab6beb67db19711fc073eefd